src/keepers/liquidation.ts

```diff
@@ -149,49 +177,48 @@ export class LiquidationKeeper extends Keeper {
   }
 
   private async liquidatePosition(account: string) {
-    const canLiquidateOrder = await this.market.canLiquidate(account);
-    if (!canLiquidateOrder) {
-      // if it's not liquidatable update it's liquidation price
-      this.positions[account].liqPrice = parseFloat(
-        utils.formatUnits((await this.market.liquidationPrice(account)).price)
-      );
-      this.positions[account].liqPriceUpdatedTimestamp = this.blockTipTimestamp;
-      this.logger.info('Cannot liquidate position', {
-        args: { account, liqPrice: this.positions[account].liqPrice },
-      });
-      return;
-    }
-
     try {
+      this.logger.info('Checking if can liquidate', { args: { account } });
+      const canLiquidateOrder = await this.market.canLiquidate(account);
+      if (!canLiquidateOrder) {
+        // if it's not liquidatable update it's liquidation price
+        this.positions[account].liqPrice = parseFloat(
+          utils.formatUnits((await this.market.liquidationPrice(account)).price)
+        );
+        this.positions[account].liqPriceUpdatedTimestamp = this.blockTipTimestamp;
+        this.logger.info('Cannot liquidate position', {
+          args: { account, liqPrice: this.positions[account].liqPrice },
+        });
+        return;
+      }
+
       await this.signerPool
         .withSigner(
           async signer => {
+            const market = this.market.connect(signer);
+
             this.logger.info('Flagging position...', { args: { account } });
-            const tx: TransactionResponse = await this.market.connect(signer).flagPosition(account);
+            const flagTx: TransactionResponse = await market.connect(signer).flagPosition(account);
             this.logger.info('Submitted transaction, waiting for completion...', {
-              args: { account, nonce: tx.nonce },
+              args: { account, nonce: flagTx.nonce },
             });
-            await this.waitTx(tx);
+            await this.waitTx(flagTx);
```

Comment:
I'm curious if this wait is required, could `flagPosition` and `liquidatePosition` be done in a multicall? No changes required, just me being curious

Reply:
Yeah, 💯 can. Most of the time liquidations are instantaneously executed, immediately after the flag and in the same tx. However, if we use the multicall, sUSD rewards will be sent to the multicall contract. So, this requires deploying a new contract and I didn't want to do that right now.